docs/archive-data.md

```diff
@@ -0,0 +1,314 @@
+
+At the cluster level, `archival.history.state` must be set to `"enabled"` for Archival to work at the Namespace level for Even Histories. Once enabled at the cluster level, a Namespace can safely switch between `enabled` and `disabled` states.
+
+Archival is supported in global Namespaces (Namespaces spanning multiple clusters). When Archival is running in a global Namepsace, it will first run on the active cluster and then some time later it will run on the standby cluster. Before archiving, a history check is done to see what has been previously uploaded.
+
+## Archive reads
+ 
+To retrieve archived Event Histories you will need the Workflow Id and the Run Id:
+
+```bash
+$ ./temporal --ns <namespace> wf show --wid <workflowId> --rid <runId>
+```
+
+:::tip
+
+Want to test this feature locally? Start by running a Temporal server:
+
+```bash
+$ ./temporal-server start
+```
+
+Then register a new Namespace with Archival enabled. 
+
+```bash
+$ ./tctl --ns samples-namespace namespace register --gd false --history_archival_status enabled --retention 0
+```
+
+Next, run a sample workflow such as the [helloworld temporal sample](https://github.com/temporalio/temporal-go-samples/tree/master/helloworld).
+
+Once the sample execution has finished, you can view the archived Event Histories by copying the `workflowId` and `runId` of the completed Workflow from the log output and running:
+
+```bash
+$ ./temporal --ns samples-namespace wf show --wid <workflowId> --rid <runId>
+```
+
+:::
+
+## Custom archiver
+
+Temporal does not intend to limit you to specific archive locations. To use a provider that is not currently supported you may create your own archiver.
+
+### Create a new package
+
+The first step is to create a new package for your implementation in [`/common/archiver`](https://github.com/temporalio/temporal/tree/master/common/archiver). Create a new directory in the archiver folder and arrange the  structure to look like the following:
+
+```
+./common/archiver
+  - filestore/                      -- Filestore implementation
+  - provider/
+      - provider.go                 -- Provider of archiver instances
+  - yourImplementation/
+      - historyArchiver.go          -- HistoryArchiver implementation
+      - historyArchiver_test.go     -- Unit tests for HistoryArchiver
+      - visibilityArchiver.go       -- VisibilityArchiver implementations
+      - visibilityArchiver_test.go  -- Unit tests for VisibilityArchiver
+```
+
+### Implement interfaces
+
+Next, implement the [`HistoryArchiver`](https://github.com/temporalio/temporal/blob/master/common/archiver/interface.go#L80) and the [`VisibilityArchiver`](https://github.com/temporalio/temporal/blob/master/common/archiver/interface.go#L109) interfaces.
+
+```go
+type HistoryArchiver interface {
+	// Archive is used to archive a workflow history. When the context expires the method should stop trying to archive.
```

Comment:
Formatting is messed up here.

Reply:
Also what file should these be in? You never say. I think putting the file name above each snippet is nice